tools/src/main/java/org/apache/kafka/trogdor/workload/ProduceBenchWorker.java

```diff
@@ -242,17 +244,21 @@ void recordDuration(long durationMs) {
         @Override
         public void run() {
             try {
-                Histogram.Summary summary = histogram.summarize(percentiles);
-                StatusData statusData = new StatusData(summary.numSamples(), summary.average(),
-                    summary.percentiles().get(0).value(),
-                    summary.percentiles().get(1).value(),
-                    summary.percentiles().get(2).value());
-                String statusDataString = JsonUtil.toJsonString(statusData);
-                status.set(statusDataString);
+                update();
             } catch (Exception e) {
                 WorkerUtils.abort(log, "StatusUpdater", e, doneFuture);
             }
         }
+
+        StatusData update() {
+            Histogram.Summary summary = histogram.summarize(percentiles);
+            StatusData statusData = new StatusData(summary.numSamples(), summary.average(),
+                summary.percentiles().get(0).value(),
```

Comment:
Would be great to make it a bit more robust w.r.t. adding/removing percentiles form Histogram, i.e.  But I see we already do this in other places. I can add JIRA to improve that?

Reply:
Hmm, Histogram is internally synchronized, so there should be no conflict.  Maybe I misunderstood the question

Reply:
Oh I meant that we explicitly get first three values from percentiles

Reply:
Actually, it's fine, but I think we should put a comment in StatusUpdater constructor that changing percentiles also need to make sure that it's consistent with json properties in StatusData and how we construct it.  The reason is that I just noticed that we actually create these percentiles: 'this.percentiles = new float[] {0.50f, 0.95f, 0.99f};` but StatusData has `@JsonProperty("p90LatencyMs") int p90latencyMs` instead of p95.

Reply:
Good catch, @apovzner .  I fixed the discrepancy.  I also moved the array into the StatusUpdater class and added a comment.